src/modal-checkout/modal.js

```diff
@@ -95,16 +108,21 @@ domReady( () => {
 				}
 				window?.newspackReaderActivation?.resetCheckoutData?.();
 			};
+
 			if ( window?.newspackReaderActivation?.openNewslettersSignupModal ) {
 				window.newspackReaderActivation.openNewslettersSignupModal( {
 					callback: handleCheckoutComplete,
+					skipClose: ( afterSuccessUrlInput && afterSuccessBehaviorInput ) ? true : false,
```

Comment:
If you decide to go with the strategy above for `shouldCloseModal`:

```suggestion
					skipClose: ! shouldCloseModal,
```

Reply:
Ah this is much nicer!